mlx/backend/cuda/allocator.cpp

```diff
@@ -157,17 +157,11 @@ CudaAllocator::malloc_async(size_t size, int device, cudaStream_t stream) {
       cudaError_t err;
       void* data = nullptr;
       if (device == -1) {
-        err = cudaMallocManaged(&data, size);
+        CHECK_CUDA_ERROR(cudaMallocManaged(&data, size));
       } else {
-        err = cudaMallocAsync(&data, size, stream);
-      }
-      if (err != cudaSuccess && err != cudaErrorMemoryAllocation) {
-        throw std::runtime_error(fmt::format(
-            "cudaMallocManaged failed: {}.", cudaGetErrorString(err)));
-      }
-      if (!data) {
-        return Buffer{nullptr};
+        CHECK_CUDA_ERROR(cudaMallocAsync(&data, size, stream));
```

Comment:
Does cudaMallocAsync return an error when the allocation fails? It might just return a nullptr without an error?